### ISSUE TITLE
Static Interface since Android N

### ISSUE THREAD

**Issue by P-Zenker** [mapbox/mapbox-java#782]
Hey,

I've updated to com.mapbox.mapboxsdk:mapbox-android-navigation-ui:0.12.0 which uses your latest
v3.0.1:

```
Static interface methods are only supported starting with Android N (--min-api 24): com.mapbox.geojson.Geometry com.mapbox.geojson.Geometry.fromJson(java.lang.String)
Message{kind=ERROR, text=Static interface methods are only supported starting with Android N (--min-api 24): com.mapbox.geojson.Geometry com.mapbox.geojson.Geometry.fromJson(java.lang.String), sources=[Unknown source file], tool name=Optional.of(D8)}
```

My build.grade looks like this:

```
apply plugin: 'com.android.application'

android {
    compileSdkVersion 27
    buildToolsVersion '27.0.3'
    flavorDimensions "default"
    defaultConfig {
        applicationId "android.catchup_apps.com.cumapboxnavigationui"
        minSdkVersion 19
        targetSdkVersion 27
        versionCode 1
        versionName "1.0"
        testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
        multiDexEnabled true
    }

    dexOptions {
        javaMaxHeapSize "4g"
    }
}
```

To make gradle compiling (without the error above) I have to specify
```
compileOptions {
        sourceCompatibility JavaVersion.VERSION_1_8
        targetCompatibility JavaVersion.VERSION_1_8
    }
```

Maybe you should update the docs or downgrade to Java 1.7 ([because not every API is supported for Android < 24](https://developer.android.com/studio/write/java8-support.html#supported_features))

**Comment by danesfeder** [comment]
Hey @P-Zenker, Sorry for the confusion surrounding Java 8. We are going to update our navigation release notes to make this more clear. 

The static interface method used in `com.mapbox.geojson.Geometry` is compatible with any minSdkVersion (see Supported Java 8 Language Features and APIs in the [article you referenced](https://developer.android.com/studio/write/java8-support.html#supported_features)).  

Also, noting that with Android Studio 3.1.0, you should not run into this issue, as the new dex compiler [`D8`](https://android-developers.googleblog.com/2017/08/next-generation-dex-compiler-now-in.html) will be enabled by default.

**Comment by danesfeder** [comment]
Going to go ahead and close this @P-Zenker, thanks for the quick 👀 on the new release.  Feel free to re-open if you run into anything else.

**Comment by LouisCAD** [comment]
@danesfeder With Android Studio 3.1.2, compilation fails with the following error plus giant stacktraces:
```console
AGPBI: {"kind":"error","text":"Static interface methods are only supported starting with Android N (--min-api 24): com.mapbox.geojson.Geometry com.mapbox.geojson.Geometry.fromJson(java.lang.String)","sources":[{}],"tool":"D8"}
```

**Comment by LouisCAD** [comment]
I do not want to set java language level to 1.8 because I don't need it, I use Kotlin, and enabling it allows me to write code that would crash on pre-nougat devices, like the infamous map iteration:
```kotlin
mapOf("" to "").forEach { k, v -> println("$k -> $v") }`
```

**Comment by zugaldia** [comment]
@LouisCAD thanks for the feedback and for using the library, also glad to see you're using Kotlin in your projects. We're trying to incrementally add more Kotlin to our projects as well, next in the form of extensions, but we're being very careful of doing so as it comes with the cost of an increase in the size of the resulting library binary. The size of the library binary is very important to a good number of our customers and we try to keep it as small as possible. For that reason, and in order to take advantage of some more modern language features, we've introduce Java 8 support as this is the main language this library is developed with.

We've been careful of not making this change to quickly in order to minimize negative impact on our clients (Java 8 support for Android was announced by Google [more than a year ago](https://android-developers.googleblog.com/2017/03/future-of-java-8-language-feature.html)). In any case, with Kotlin being an officially supported language (the default language nowadays I'd say) I can imagine that we're gonna see less and less issues like the one you report. Meanwhile, I 
recommend you to [submit a bug upstream](https://developer.android.com/studio/write/java8-support#disable) (if there isn't one yet) and keep sharing your feedback with us.

**Comment by LouisCAD** [comment]
You got it wrong. Android Gradle Plugin 3.0+ desugars Java 8 language
features like default interface method implementations, so pre-N devices
can run your app using Java 8 language features. Kotlin works starting with
JDK 6, and you can use JDK 7 safely on all Android versions with it.

**Comment by santa010** [comment]
But if I understand correctly, the Android Studio 3.1 still needs the
```
compileOptions {
        sourceCompatibility JavaVersion.VERSION_1_8
        targetCompatibility JavaVersion.VERSION_1_8
    }

```
to work?

**Comment by paulkugaev** [comment]
specifying java version does not work wor me

**Comment by zugaldia** [comment]
@santa010 @paulkugaev If you're still seeing issues after applying the `compileOptions` suggested in OP, could you [cut a new ticket](https://github.com/mapbox/mapbox-java/issues/new) and include any error messages you're getting? Thanks!

**Comment by Ni3Narale** [comment]
android {
    compileSdkVersion 26
    buildToolsVersion '27.0.3'

    defaultConfig {
        vectorDrawables.useSupportLibrary = true
        applicationId "in.myapp.app"
        minSdkVersion 17
        targetSdkVersion 26
        versionCode 32
        versionName "1.0.32"
        multiDexEnabled true
    }
    buildTypes {
        release {
            minifyEnabled false
            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
        }
    }

    packagingOptions {
        exclude 'META-INF/NOTICE'
        exclude 'META-INF/LICENSE'
        exclude 'META-INF/notice'
        exclude 'META-INF/notice.txt'
        exclude 'META-INF/license'
        exclude 'META-INF/license.txt'
    }
    dexOptions {
        javaMaxHeapSize "Xg"
    }
    lintOptions {
        abortOnError false
    }
    buildTypes {
        release {
            minifyEnabled false
            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
        }
        debug {
            debuggable true
        }
    }
}

dependencies {
    implementation fileTree(dir: 'libs', include: ['*.jar'])
    implementation 'com.android.support.constraint:constraint-layout:1.0.2'
    testImplementation 'junit:junit:4.12'
    androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.1'
    //noinspection GradleCompatible
    implementation 'com.android.support:design:26.1.0'
    implementation 'com.android.volley:volley:1.0.0'
    implementation 'com.google.code.gson:gson:2.3'
    implementation 'com.android.support:multidex:1.0.1'
    implementation 'de.hdodenhof:circleimageview:1.2.1'
//    implementation 'com.intuit.sdp:sdp-android:1.0.5'
    implementation 'com.github.gcacace:signature-pad:1.2.1'
    implementation 'com.android.support:support-v4:26.1.0'
    implementation 'com.android.support:appcompat-v7:26.+'
    implementation 'com.android.support:recyclerview-v7:26.1.0'
    implementation 'com.android.support:cardview-v7:26.1.0'
    implementation 'com.soundcloud.android:android-crop:1.0.1@aar'
    implementation 'com.github.bumptech.glide:glide:3.5.2'
    implementation 'com.google.android.gms:play-services-location:11.4.0'
    implementation 'com.google.android.gms:play-services-maps:11.4.0'
    implementation 'com.google.android.gms:play-services:11.4.0'
    implementation 'com.google.android.gms:play-services-gcm:11.4.0'
    implementation 'com.google.maps.android:android-maps-utils:0.5'
    implementation 'com.squareup.retrofit2:retrofit:2.3.0'
    implementation 'com.squareup.retrofit2:converter-scalars:2.3.0'
    implementation 'com.squareup.retrofit2:converter-gson:2.3.0'
    implementation 'com.squareup.picasso:picasso:2.5.2'
    implementation 'com.google.firebase:firebase-core:11.4.0'
    implementation 'com.google.firebase:firebase-messaging:11.4.0'
    implementation 'com.intuit.sdp:sdp-android:1.0.5'
    implementation 'com.shuhart.stepview:stepview:1.2.6'
    implementation 'com.xw.repo:bubbleseekbar:3.6'
    implementation 'com.github.andyxialm:ColorDialog:1.0.0'
    //implementation 'com.crashlytics.sdk.android:crashlytics:2.9.3'
}
apply plugin: 'com.google.gms.google-services'



Using this app gradel for application due to change
implementation 'com.android.support:design:27.1.1'

need to change all implemetation in app

give me suggetion for same

**Comment by paulkugaev** [comment]
@zugaldia actually the main issue was the newest R8 in android studio 3.3. I've already filled a form on their issue tracker

**Comment by osana** [comment]
@paulkugaev @Ni3Narale we recently released `4.0.0` that is DOES NOT require java 8 support. 
Could you see if you still see your problems running against mapbox-java `4.0.0`
